### PR TITLE
[feat]: [CDS-77390]: Exposed CRUD operations for gitx webhooks

### PR DESCRIPTION
**Pull request by adivishy1**
## Describe your changes
Added new api contracts for performing CRUD operations for git webhooks in the open api spec.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51970)

### CODE REVIEW

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -1506,6 +1506,86 @@ paths:
       tags:
         - EULA
       x-internal: false
+  /v1/gitx-webhooks:
+    post:
+      summary: Create a GitX webhook
+      operationId: create-gitx-webhooks
+      responses:
+        '200':
```
Comment (mankrit-singh) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:1514]: if this is creating an entity, please make the code as 201

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -1506,6 +1506,86 @@ paths:
...
+      requestBody:
+        $ref: '#/components/requestBodies/UpdateGitXWebhookRequest'
+      x-internal: true
+      parameters:
+        - $ref: '#/components/parameters/AccountHeader'
+    delete:
+      summary: Deletes a GitX Webhook
+      operationId: delete-gitx-webhook
+      responses:
+        '200':
```
Comment (mankrit-singh) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:1579]: please make this as 204

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -2525,6 +2605,105 @@ components:
...
+      x-stoplight:
+        id: jpe7hhufol4wp
+      type: object
+      description: 'Contains information about the GitX webhook updation '
+      properties:
+        folder_paths:
+          type: array
+          items:
+            type: string
+        isEnabled:
```
Comment (mankrit-singh) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:2645]: casing

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 4]
```diff
@@ -2525,6 +2605,105 @@ components:
...
+          type: string
+        connector_ref:
+          type: string
+        repo_name:
+          type: string
+        folder_paths:
+          type: array
+          items:
+            type: string
+        isEnabled:
```
Comment (mankrit-singh) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:2697]: casing here too

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml [hunk 5]
```diff
@@ -3119,6 +3298,36 @@ components:
...
+        application/json:
+          schema:
+            $ref: '#/components/schemas/DeleteGitXWebhookResponse'
+    GetGitXWebhookResposne:
+      description: Example response
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/GetGitXWebhookResposne'
+    ListGitXWebhookResponse:
```
Comment (mankrit-singh) [inline 120-ng-manager/contracts/openapi/v1/openapi.yaml:3325]: please add pagination headers like in List Pipelines, thanks!

Comment (adivishy1) [reply 120-ng-manager/contracts/openapi/v1/openapi.yaml:3325]: Done

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml (hunk 4 above)
Comment (adivishy1) [reply 120-ng-manager/contracts/openapi/v1/openapi.yaml:2697]: Done

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml (hunk 3 above)
Comment (adivishy1) [reply 120-ng-manager/contracts/openapi/v1/openapi.yaml:2645]: Done

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml (hunk 2 above)
Comment (adivishy1) [reply 120-ng-manager/contracts/openapi/v1/openapi.yaml:1579]: Done

File: 120-ng-manager/contracts/openapi/v1/openapi.yaml (hunk 1 above)
Comment (adivishy1) [reply 120-ng-manager/contracts/openapi/v1/openapi.yaml:1514]: Done

**Review by mankrit-singh — APPROVED** [review]

**Comment by adivishy1** [comment]
trigger uts

**Comment by adivishy1** [comment]
trigger ut2, ut9

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51970)* status: 2 file(s) to review. 1 change requests. 5. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/gitxwebhook/GitXWebhooksApiImpl.java` line 25 at r3](https://reviewable.io/reviews/harness/harness-core/51970#-Ncg5pV21r3hvd16-SB2:-Ncg5pV39HjNXR-uaZSO:bos5m0s) ([raw file](https://github.com/harness/harness-core/blob/e2e26f519c5f50f840adde941e58f4bbaa1367d6/120-ng-manager/src/main/java/io/harness/ng/gitxwebhook/GitXWebhooksApiImpl.java#L25)):*
> ```Java
> public class GitXWebhooksApiImpl implements GitXWebhooksApi {
>   @Override
>   public Response createGitxWebhooks(@Valid CreateGitXWebhookRequest body, String harnessAccount) {
> ```

shouldn't it be "createGitXWebhook" (last "s" shouldn't be there)?

**Comment by mohitg0795** [comment]
*[`120-ng-manager/src/main/java/io/harness/ng/gitxwebhook/GitXWebhooksApiImpl.java` line 30 at r3](https://reviewable.io/reviews/harness/harness-core/51970#-Ncg5wm_8Io9Sksqe007:-Ncg5wm_8Io9Sksqe008:b-2guby3) ([raw file](https://github.com/harness/harness-core/blob/e2e26f519c5f50f840adde941e58f4bbaa1367d6/120-ng-manager/src/main/java/io/harness/ng/gitxwebhook/GitXWebhooksApiImpl.java#L30)):*
> ```Java
> 
>   @Override
>   public Response getGitxWebhooks(String gitXWebhookIdentifier, String harnessAccount) {
> ```

same here

**Review by adivishy1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51970)* status: 2 file(s) to review. 1 change requests. 7. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/gitxwebhook/GitXWebhooksApiImpl.java` line 25 at r3](https://reviewable.io/reviews/harness/harness-core/51970#-Ncg5pV21r3hvd16-SB2:-NcgAgIE6hXKvvrwBBNs:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e2e26f519c5f50f840adde941e58f4bbaa1367d6/120-ng-manager/src/main/java/io/harness/ng/gitxwebhook/GitXWebhooksApiImpl.java#L25)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

shouldn't it be "createGitXWebhook" (last "s" shouldn't be there)?
</blockquote></details>

Done.
___
*[`120-ng-manager/src/main/java/io/harness/ng/gitxwebhook/GitXWebhooksApiImpl.java` line 30 at r3](https://reviewable.io/reviews/harness/harness-core/51970#-Ncg5wm_8Io9Sksqe007:-NcgAg4R1Hh0Iu6H6jH2:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e2e26f519c5f50f840adde941e58f4bbaa1367d6/120-ng-manager/src/main/java/io/harness/ng/gitxwebhook/GitXWebhooksApiImpl.java#L30)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.

**Comment by adivishy1** [comment]
trigger uts

**Review by mankrit-singh — APPROVED** [review]

**Review by mohitg0795 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51970)* status: 2 file(s) to review. 5. unresolved discussions.

**Comment by adivishy1** [comment]
trigger review

**Pull request merged**